src/identity/group.rs

```diff
@@ -0,0 +1,126 @@
+use crate::{
+    api::{
+        self,
+        identity::group::{
+            requests::{
+                CreateGroupByNameRequest, CreateGroupByNameRequestBuilder, CreateGroupRequest,
+                CreateGroupRequestBuilder, DeleteGroupByIdRequest, DeleteGroupByNameRequest,
+                ListGroupsByIdRequest, ListGroupsByNameRequest, ReadGroupByIdRequest,
+                ReadGroupByNameRequest, UpdateGroupByIdRequest, UpdateGroupByIdRequestBuilder,
+            },
+            responses::{
+                CreateGroupResponse, ListGroupsByIdResponse, ListGroupsByNameResponse,
+                ReadGroupByIdResponse, ReadGroupByNameResponse,
+            },
+        },
+    },
+    client::Client,
+    error::ClientError,
+};
+
+/// Creates or update a group.
+///
+/// See [CreateGroupRequest]
+#[instrument(skip(client, opts), err)]
+pub async fn create(
+    client: &impl Client,
+    opts: Option<&mut CreateGroupRequestBuilder>,
+) -> Result<Option<CreateGroupResponse>, ClientError> {
+    let mut t = CreateGroupRequest::builder();
+    let endpoint = opts.unwrap_or(&mut t).build().unwrap();
+    api::exec_with_result_or_empty(client, endpoint).await
```

Comment:
This uses the `UpdateGroupByIdRequest` if the `CreateGroupRequest` contains an id
```suggestion
    if endpoint.id.is_some() {
        let update_endpoint: UpdateGroupByIdRequest = endpoint.into();
        let _ = api::exec_with_empty(client, update_endpoint).await?;
        Ok(None)
    } else {
        Ok(Some(api::exec_with_result(client, endpoint).await?))
    }
```